src/fields/Fields.cpp

```diff
@@ -493,27 +506,85 @@ void
 Fields::SetBoundaryCondition (amrex::Vector<amrex::Geometry> const& geom, const int lev,
                               std::string component, const int islice)
 {
-    if (lev == 0) return; // keep lev==0 boundaries zero
     HIPACE_PROFILE("Fields::SetBoundaryCondition()");
-    constexpr int interp_order = 2;
+    if (lev == 0 && m_open_boundary) {
+        // Coarsest level: use Taylor expansion of the Green's function
+        // to get Dirichlet boundary conditions
 
-    const amrex::Real ref_ratio_z = Hipace::GetRefRatio(lev)[2];
-    const amrex::Real islice_coarse = (islice + 0.5_rt) / ref_ratio_z;
-    const amrex::Real rel_z = islice_coarse - static_cast<int>(amrex::Math::floor(islice_coarse));
+        amrex::MultiFab staging_area = getStagingArea(lev);
+        // Open Boundaries only work for lev0 with everything in one box
```

Comment:
If there are some options incompatible with open BC, could you write corresponding asserts?

Reply:
Done